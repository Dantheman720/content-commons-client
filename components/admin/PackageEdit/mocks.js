import {
  tag1, tag2, mocks as pkgFiles
} from 'components/admin/PackageEdit/PackageFiles/mocks';
import { PACKAGE_QUERY, DELETE_PACKAGE_MUTATION } from 'lib/graphql/queries/package';

const { documents } = pkgFiles[0].result.data.pkg;

const pressJournalism = {
  __typename: 'Category',
  id: 'ck2lzgu1e0red072066m25ldt',
  translations: [
    {
      __typename: 'LanguageTranslation',
      id: 'ck2lzfz0d0i580720g3mg0xut',
      name: 'press & journalism',
      language: {
        __typename: 'Language',
        id: 'ck2lzfx710hkq07206thus6pt',
        languageCode: 'en',
        locale: 'en-us',
        textDirection: 'LTR',
        displayName: 'English',
        nativeName: 'English'
      }
    },
    {
      __typename: 'LanguageTranslation',
      id: 'ck2lzfz0y0i5f0720az8sehv4',
      name: 'Prensa y periodismo',
      language: {
        __typename: 'Language',
        id: 'ck2lzfx7o0hl707205uteku77',
        languageCode: 'es',
        locale: 'es-es',
        textDirection: 'LTR',
        displayName: 'Spanish',
        nativeName: 'Español'
      }
    },
    {
      __typename: 'LanguageTranslation',
      id: 'ck2lzfz1i0i5m0720wa9pemol',
      name: 'Presse et journalisme',
      language: {
        __typename: 'Language',
        id: 'ck2lzfx710hkp07206oo0icbv',
        languageCode: 'fr',
        locale: 'fr-fr',
        textDirection: 'LTR',
        displayName: 'French',
        nativeName: 'Français'
      }
    }
  ]
};

export const props = {
  router: {
    push: jest.fn(),
    query: {
      id: 'test-123'
    }
  }
};

export const mocks = [
  {
    request: {
      query: PACKAGE_QUERY,
      variables: { id: props.router.query.id }
    },
    result: {
      data: {
        pkg: {
          __typename: 'Package',
          id: props.router.query.id,
          createdAt: '2019-11-12T13:07:49.364Z',
          updatedAt: '2019-11-12T13:08:28.830Z',
          publishedAt: '',
          type: 'DAILY_GUIDANCE',
          assetPath: '',
          author: {
            __typename: 'User',
            id: 'ck2m042xo0rnp0720nb4gxjix',
            firstName: 'First',
            lastName: 'Last'
          },
          team: {
            __typename: 'Team',
            id: 'ck2qgfbku0ubh0720iwhkvuyn',
            name: 'GPA Press Office'
          },
          title: 'Final Guidance mm-dd-yy',
          desc: '',
          status: 'DRAFT',
          visibility: 'INTERNAL',
          categories: [pressJournalism],
          tags: [tag1, tag2],
          documents
        }
      }
    }
  },
  {
    request: {
      query: DELETE_PACKAGE_MUTATION,
      variables: { id: props.router.query.id }
    },
    result: {
      data: {
        deletePackage: {
          __typename: 'Package',
          id: props.router.query.id
        }
      }
    }
  },
  // {
  //   request: {
  //     query: 'UPDATE_PACKAGE_MUTATION',
  //     variables: {
  //       data: { title: 'New Title' },
  //       where: { id: props.router.query.id }
  //     }
  //   },
  //   result: {
  //     data: {
  //       updatePackage: {
  //         __typename: 'Package',
  //         id: props.router.query.id,
  //         title: 'New Title'
  //       }
  //     }
  //   }
  // }
];

export const errorMocks = [
  {
    ...mocks[0],
    result: {
      errors: [{ message: 'There was an error.' }]
    }
  },
  { ...mocks[1] }
];

export const publishedMocks = [
  {
    ...mocks[0],
    result: {
      data: {
        package: {
          ...mocks[0].result.data.package,
          publishedAt: '2019-11-13T13:08:28.830Z'
        }
      }
    }
  },
  {
    request: {
      query: 'DELETE_PACKAGE_MUTATION',
      variables: { id: props.router.query.id }
    },
    result: {
      data: {
        deletePackage: {
          __typename: 'Package',
          id: props.router.query.id
        }
      }
    }
  },
  // {
  //   request: {
  //     query: 'UPDATE_PACKAGE_MUTATION',
  //     variables: {
  //       data: { title: 'New Title' },
  //       where: { id: props.router.query.id }
  //     }
  //   },
  //   result: {
  //     data: {
  //       updatePackage: {
  //         __typename: 'Package',
  //         id: props.router.query.id,
  //         title: 'New Title'
  //       }
  //     }
  //   }
  // }
];

export const noDocumentsMocks = [
  {
    ...mocks[0],
    result: {
      data: {
        pkg: {
          ...mocks[0].result.data.pkg,
          documents: []
        }
      }
    }
  },
  { ...mocks[1] }
];
