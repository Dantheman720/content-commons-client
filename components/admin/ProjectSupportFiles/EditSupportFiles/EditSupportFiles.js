import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import {
  Grid, Form, Button, Modal, Header, Dimmer
} from 'semantic-ui-react';
import ButtonAddFiles from 'components/ButtonAddFiles/ButtonAddFiles';
import { useFileActions } from 'lib/hooks/useFileActions';
import Notification from 'components/Notification/Notification';
import EditSupportFileRow from '../EditSupportFileRow/EditSupportFileRow';
import './EditSupportFiles.scss';

const EditSupportFiles = ( {
  supportFiles, extensions, save
} ) => {
  const [open, setOpen] = useState( false );
  const [saving, setSaving] = useState( false );
  const [allFieldsSelected, setAllFieldsSelected] = useState( false );

  const {
    state: { files, filesToRemove },
    reset,
    updateFileField,
    addFiles,
    removeFile,
    replaceFile
  } = useFileActions();

  const notificationStyles = {
    position: 'absolute',
    top: '15px',
    right: '40px'
  };


  /*
  Check to see if all required dropdowns are completed
  when the the files state changes. All fields do not need
  to be checked as some are pre-populated on initialization or
  not applicable. Save button becomes active when all
  complete
  */
  const isComplete = () => files.every( file => !!file.language );

  useEffect( () => {
    setAllFieldsSelected( isComplete() );
  }, [files] );


  const allowedExtensions = extensions.join( ',' );
  const compareFilenames = ( a, b ) => {
    try {
      return a.name.localeCompare( b.name );
    } catch ( err ) {
      console.log( err );
    }
  };

  const closeModal = () => {
    setOpen( false );
    reset();
  };

  const openModal = () => {
    addFiles( supportFiles );
    setOpen( true );
  };

  const updateField = ( e, data ) => {
    updateFileField( data );
  };

  const handleSave = async () => {
    setSaving( true );
    await save( files, filesToRemove );
    setSaving( false );
    closeModal();
  };

  return (

    <Modal
      open={ open }
      onClose={ closeModal }
      className="edit-support-files"
      size="small"
      trigger={ <Button className="btn--edit" onClick={ openModal } size="small" basic>Edit</Button> }
    >
      <Header content="Edit thumbnail files in this project" />

      { saving
      && (
      <Notification
        el="p"
        show
        customStyles={ notificationStyles }
        msg="Saving changes..."
      />
      )
    }
      <Dimmer.Dimmable dimmed={ saving }>
        <Dimmer inverted simple />

        <Modal.Content>
          <Form>
            <Grid>
              <Grid.Row>
                <Grid.Column width={ 8 }>Files Selected</Grid.Column>
                <Grid.Column width={ 6 }>Language<span className="required"> *</span></Grid.Column>
                <Grid.Column width={ 2 }></Grid.Column>
              </Grid.Row>

              { files.sort( compareFilenames ).map( file => (
                <EditSupportFileRow
                  key={ file.id }
                  file={ file }
                  update={ updateField }
                  removeFile={ removeFile }
                  replaceFile={ replaceFile }
                  accept={ allowedExtensions }
                />
              ) ) }
            </Grid>
          </Form>
        </Modal.Content>
        <Modal.Actions>
          <Button
            className="secondary alternative"
            content="Cancel"
            onClick={ closeModal }
          />
          <ButtonAddFiles
            onChange={ e => addFiles( e.target.files ) }
            multiple
            accept={ allowedExtensions }
            className="secondary"
          >Add Files
          </ButtonAddFiles>

          <Button
            className="primary"
            type="button"
            content="Save"
            onClick={ handleSave }
            disabled={ !allFieldsSelected }
          />
        </Modal.Actions>

      </Dimmer.Dimmable>
    </Modal>
  );
};

EditSupportFiles.propTypes = {
  supportFiles: PropTypes.array,
  extensions: PropTypes.array,
  save: PropTypes.func
};

export default EditSupportFiles;
