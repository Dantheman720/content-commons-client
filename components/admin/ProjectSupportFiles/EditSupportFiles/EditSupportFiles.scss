@import "styles/colors.scss";

.edit-support-files {
  &.ui.modal.small {
    width: 70%;
    padding-bottom: 20px;

    @media only screen and (min-width: 992px) {
      width: 585px;
    }

    @media only screen and (min-width: 1200px) {
      width: 625px;
    }

    .ui.inverted.simple.dimmer {
      background-color: rgba(255, 255, 255, 0.3);
    }

    .ui.header {
      background: #fff;
      border-bottom: transparent;
      padding: 1.25rem 1.5rem 1em 1.5rem;
      color: $blue;
    }

    .content {
      padding-top: 10px;

      .ui.grid {
        margin-left: 0.5rem;
        margin-right: 0.5rem;

        .row {
          &:first-child {
            border-bottom: 2px solid $alt-grey;
            font-size: 1rem;
          }

          &:nth-child(2) {
            padding-top: 15px;
          }
          padding-bottom: 0.5rem;
          padding-top: 0.5rem;
          font-size: 0.888888889rem;
        }
      }
    }

    .actions {
      background: #fff;
      border-top: transparent;
      padding-top: 2.2em;
      text-align: center;
    }

    .required {
      color: $red;
    }
  }
}
