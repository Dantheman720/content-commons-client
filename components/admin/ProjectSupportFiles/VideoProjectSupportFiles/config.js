export const config = {
  supportFiles: {
    // query: VIDEO_PROJECT_QUERY,
    types: {
      srt: {
        headline: 'SRT Files',
        extensions: ['.srt'],
        popupMsg: 'Some info about what SRT files are.'
      },
      other: {
        headline: 'Additional Files',
        extensions: ['.png', '.jpeg', '.jpg', '.svg'],
        popupMsg: 'Additional files that can be used with this video, e.g., audio file, pdf.',
        iconMsg: 'Checking this prevents people from downloading and using your images. Useful if your images are licensed.',
        iconSize: 'small',
        iconType: 'info circle'
      }
    }
  }
};
