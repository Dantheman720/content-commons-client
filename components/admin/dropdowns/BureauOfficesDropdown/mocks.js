export const bureaus = [
  {
    id: 'sdfq',
    name: 'Bureau 1',
    abbr: 'B1',
    offices: [
      {
        id: 'kglf',
        name: 'Office 1',
        abbr: 'O1'
      },
      {
        id: 'eiwo',
        name: 'Office 2',
        abbr: 'O2'
      }
    ]
  },
  {
    id: 'weio',
    name: 'Bureau 2',
    abbr: 'B2',
    offices: [
      {
        id: 'eiwo',
        name: 'Office 2',
        abbr: 'O2'
      }
    ]
  },
  {
    id: 'xzwi',
    name: 'Bureau 3',
    abbr: 'B3',
    offices: [
      {
        id: 'kglf',
        name: 'Office 1',
        abbr: 'O1'
      }
    ]
  },
  {
    id: 'zxcw',
    name: 'Bureau 4',
    abbr: 'B4',
    offices: [
      {
        id: 'kglf',
        name: 'Office 1',
        abbr: 'O1'
      },
      {
        id: 'eiwo',
        name: 'Office 2',
        abbr: 'O2'
      }
    ]
  }
];
