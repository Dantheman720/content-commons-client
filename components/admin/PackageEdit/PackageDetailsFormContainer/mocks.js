import { mocks as pkgFiles } from 'components/admin/PackageEdit/PackageFiles/mocks';
import { language } from 'components/admin/PackageEdit/mocks';
import { PACKAGE_QUERY } from 'lib/graphql/queries/package';

const { documents } = pkgFiles[0].result.data.package;

const tag1 = {
  __typename: 'Tag',
  id: 'ck2lzgu1i0rei07206gvy1ygg',
  translations: [
    {
      __typename: 'LanguageTranslation',
      id: 'ck2lzfzwr0iey0720hrigffxo',
      name: 'american culture',
      language
    }
  ]
};

const tag2 = {
  __typename: 'Tag',
  id: 'ck2lzgu2s0rer07208jc6y6ww',
  translations: [
    {
      __typename: 'LanguageTranslation',
      id: 'ck2lzg1900iui0720q28le4rs',
      name: 'leadership',
      language
    }
  ]
};

/**
 * mock data for UI development
 * eventually use for unit testing
 */
export const props = { id: 'test-123' };

export const mocks = [
  {
    request: {
      query: PACKAGE_QUERY,
      variables: { id: props.id }
    },
    result: {
      data: {
        package: {
          __typename: 'Package',
          id: props.id,
          createdAt: '2019-11-12T13:07:49.364Z',
          updatedAt: '2019-11-12T13:08:28.830Z',
          publishedAt: '',
          type: 'DAILY_GUIDANCE',
          title: 'Final Guidance mm-dd-yy',
          desc: '',
          visibility: 'INTERNAL',
          tags: [tag1, tag2],
          documents
        }
      }
    }
  },
  {
    request: {
      query: 'UPDATE_PACKAGE_MUTATION',
      variables: {
        data: { title: 'New Title' },
        where: { id: props.id }
      }
    },
    result: {
      data: {
        updatePackage: {
          __typename: 'Package',
          id: props.id,
          title: 'New Title'
        }
      }
    }
  }
];
