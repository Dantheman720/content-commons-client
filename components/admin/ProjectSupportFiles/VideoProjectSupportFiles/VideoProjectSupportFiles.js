import React from 'react';
import ProjectSupportFiles from '../ProjectSupportFiles';
import { config } from './config';

const updateDatabase = files => {
  console.log( 'updateFiles' );
  console.dir( files );
};

const removeFromDataBase = files => {
  console.log( 'removeFiles' );
  console.dir( files );
};


const VideoProjectSupportFiles = props => (
  <ProjectSupportFiles
    { ...props }
    updateDatabase={ updateDatabase }
    removeFromDataBase={ removeFromDataBase }
    config={ config.supportFiles }
  />
);

export default VideoProjectSupportFiles;
