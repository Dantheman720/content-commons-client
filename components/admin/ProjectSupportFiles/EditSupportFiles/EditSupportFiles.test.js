import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { data } from '../SupportFileTypeList/mocks';
import { config } from '../VideoProjectSupportFiles/config';
import EditSupportFiles from './EditSupportFiles';

const props = {
  supportFiles: data.projectFiles.supportFiles,
  extensions: config.supportFiles.types.srt.extensions,
  save: jest.fn()
};

const Component = <EditSupportFiles { ...props } />;

describe( '<EditSupportFiles />', () => {
  it( 'renders without crashing', () => {
    const wrapper = shallow( Component );

    expect( wrapper.exists() ).toEqual( true );
    expect( toJSON( wrapper ) ).toMatchSnapshot();
  } );

  it( 'clicking Save Button calls props.save', () => {
    const wrapper = shallow( Component );
    const saveBtn = wrapper.find( 'Button.primary' );

    saveBtn.simulate( 'click' );
    expect( props.save ).toHaveBeenCalled();
  } );
} );
