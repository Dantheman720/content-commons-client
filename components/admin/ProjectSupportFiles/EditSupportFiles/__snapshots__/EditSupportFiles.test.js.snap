// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditSupportFiles /> renders without crashing 1`] = `
<Modal
  centered={true}
  className="edit-support-files"
  closeOnDimmerClick={true}
  closeOnDocumentClick={false}
  dimmer={true}
  eventPool="Modal"
  onClose={[Function]}
  open={false}
  size="small"
  trigger={
    <Button
      as="button"
      basic={true}
      className="btn--edit"
      onClick={[Function]}
      size="small"
    >
      Edit
    </Button>
  }
>
  <Header
    content="Edit thumbnail files in this project"
  />
  <DimmerDimmable
    dimmed={false}
  >
    <Dimmer
      inverted={true}
      simple={true}
    />
    <ModalContent>
      <Form
        as="form"
      >
        <Grid>
          <GridRow>
            <GridColumn
              width={8}
            >
              Files Selected
            </GridColumn>
            <GridColumn
              width={6}
            >
              Language
              <span
                className="required"
              >
                 *
              </span>
            </GridColumn>
            <GridColumn
              width={2}
            />
          </GridRow>
        </Grid>
      </Form>
    </ModalContent>
    <ModalActions>
      <Button
        as="button"
        className="secondary alternative"
        content="Cancel"
        onClick={[Function]}
      />
      <ButtonAddFiles
        accept=".srt"
        className="secondary"
        disabled={false}
        multiple={true}
        onChange={[Function]}
      >
        Add Files
      </ButtonAddFiles>
      <Button
        as="button"
        className="primary"
        content="Save"
        disabled={true}
        onClick={[Function]}
        type="button"
      />
    </ModalActions>
  </DimmerDimmable>
  <DynamicConfirm
    cancelButton="Cancel"
    confirmButton="OK"
    content=""
    headline=""
    open={false}
  />
</Modal>
`;
